src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -991,6 +1007,12 @@
   ; gone through this function
   (map #(get-in query-rec [:projections % :field]) (sort columns)))
 
+(defn strip-function-calls
+  [column]
+  (let [split (group-by #(= "function" (first %)) column)]
```

Comment:
Or what I means is can't you just do `{functions true args false}` instead of `split`, functions and args could by nil, its the same as a get